### PR TITLE
chore(vsc): cache telemetry event during deactivation

### PR DESCRIPTION
**Pull request by tecton**
Telemetry events are not sent in `deactivate()` function. Cache it locally and resend next time.

### CODE REVIEW

**Review by chagong — APPROVED** [review]

File: packages/vscode-extension/src/telemetry/extTelemetry.ts [hunk 1]
```diff
@@ -186,6 +193,34 @@ export namespace ExtTelemetry {
...
   }
 
+  export async function cacheTelemetryEventAsync(
+    eventName: string,
+    properties?: { [p: string]: string }
+  ) {
+    const telemetryEvents = {
+      eventName: eventName,
+      properties: {
+        [TelemetryProperty.CorrelationId]: lastCorrelationId,
```
Comment (HuihuiWu-Microsoft) [inline packages/vscode-extension/src/telemetry/extTelemetry.ts:203]: Is this correlation id related to deactivation event?

Comment (tecton) [reply packages/vscode-extension/src/telemetry/extTelemetry.ts:203]: Yes, it helps to connect the deactivate event with last operation. Otherwise, the last operation has no "end" event.

**Review by HuihuiWu-Microsoft — APPROVED** [review]
